modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java

```diff
@@ -150,15 +149,30 @@ private MuleArtifactContext doCreateApplicationContext(MuleContext muleContext,
     if (enableLazyInit) {
       return new LazyMuleArtifactContext(muleContext, resolveArtifactConfigResources(), artifactDeclaration,
                                          optionalObjectsController,
-                                         getArtifactProperties(), artifactType, resolveContextArtifactPluginClassLoaders(),
+                                         getArtifactProperties(), artifactType, resolveAppAndDomainPluginClassLoaders(),
                                          resolveParentConfigurationProperties(), disableXmlValidations);
     }
 
     return new MuleArtifactContext(muleContext, resolveArtifactConfigResources(), artifactDeclaration, optionalObjectsController,
-                                   getArtifactProperties(), artifactType, resolveContextArtifactPluginClassLoaders(),
+                                   getArtifactProperties(), artifactType, resolveAppAndDomainPluginClassLoaders(),
                                    resolveParentConfigurationProperties(), disableXmlValidations);
   }
 
+  private List<ClassLoader> resolveAppAndDomainPluginClassLoaders() {
```

Comment:
I'm failing to see how this method return the Domain plugin classloaders. Can you explain how that works?

Reply:
I renamed the method to resolveArtifactAndParentContextsPluginClassLoaders() since it's also executed for policies and apps.

For what i could see, the imported static method (resolveContextArtifactPluginClassLoaders()) calls "getArtifactPluginClassLoaders" on the current thread contextClassLoader. Getting the pluginClassLoaders defined when that particular contextClassLoader was created. (Check org.mule.runtime.module.deployment.impl.internal.application.DefaultApplicationFactory#createArtifact).

So by getting the parentContext's classLoader and setting it as the current thread context one, i end up getting the domain class loader (when the artifact is a policy, i get the app class loader), and the static method looks for the plugins class loaders from de domain one.